repos/public-user/src/Module/SiteSettingExtension.php

```diff
@@ -0,0 +1,204 @@
+                            'label' => 'New user role', // @translate
+                            'info' => 'The user role for users when registering' // @translate
+                        ])
+                        ->setValueOptions([
+                            // This can only be transcriber for the time being.
+                            'Transcriber' => 'Transcriber'
+                        ])
+                        ->setValue($form->getSiteSettings()->get('public-user-registration-role', 'Transcriber'))
+                )
+                ->add(
+                    (new Checkbox('public-user-profile-logged-in'))
+                        ->setOptions([
+                            'label' => 'Semi-public profile', // @translate
+                            'info' => 'Only show user profiles for logged in users, otherwise for all visitors', // @translate
+                        ])
+                        ->setValue($form->getSiteSettings()->get('public-user-profile-logged-in', false))
+                )
+                ->add(
+                    (new Checkbox('public-user-profile-email'))
+                        ->setOptions([
+                            'label' => 'Show email on public profile', // @translate
+                            'info' => 'Shows the users email on their public profile', // @translate
+                        ])
+                        ->setValue($form->getSiteSettings()->get('public-user-profile-email', false))
+                )
+                ->setOptions([
+                    'label' => 'Public user options', // @translate
+                ])
+        );
+
+        // Permissions.
+        $form
+            ->add(
+                (new Fieldset('public-user-site-permissions-viewer', [
+                    'label' => 'Permissions (Role: viewers)', // @translate
+                ]))
+                    ->add($this->createCheckbox(
+                        $form,
+                        'public-user-site-permissions-viewer-view-canvas',
+                        'Can viewers see individual canvas pages' // @translate
+                    ))
+                    ->add($this->createCheckbox(
+                        $form,
+                        'public-user-site-permissions-viewer-view-manifest',
+                        'Can viewers see individual manifest pages' // @translate
+                    ))
+                    ->add($this->createCheckbox(
+                        $form,
+                        'public-user-site-permissions-viewer-view-collection',
+                        'Can viewers see individual collection pages' // @translate
+                    ))
+                    ->add($this->createCheckbox(
+                        $form,
+                        'public-user-site-permissions-viewer-view-all-collections',
+                        'Can viewers see all collections page' // @translate
+                    ))
+                ->setAttribute('id', 'viewer-permissions')
```

Comment:
@garyttierney do you think I should create a loop for these, instead of explicitly listing out the permissions. It's effectively a table of users and permissions. Each user and permission gets a checkbox. It's not the prettiest in the Omeka UI, but can possible come back to that. Was experimenting with `CheckboxList` too, which isn't styled in Omeka, but isn't too far off from being useful.